addon/modifiers/basic-dropdown-trigger.ts

```diff
@@ -0,0 +1,157 @@
+import Modifier from 'ember-modifier';
+import { action } from '@ember/object';
+import { registerDestructor } from '@ember/destroyable';
+import hasMoved from '../utils/has-moved';
+import { Dropdown } from '../components/basic-dropdown';
+
+interface NamedArgs {
+  dropdown: Dropdown;
+  eventType?: 'click' | 'mousedown';
+  stopPropagation?: boolean;
+  [named: string]: unknown;
+}
+type PositionalArgs = unknown[];
+interface Args {
+  positional: PositionalArgs;
+  named: NamedArgs;
+};
+
+export default class DropdownTriggerModifier extends Modifier<Args> {
```

Comment:
One thing I've learned only in the last couple days - the new Modifier API has a Signature type that we can leverage here for modifier args.
https://github.com/ember-modifier/ember-modifier#the-signature-type

It won't functionally affect this addon, and I'm a little busy over the Easter weekend, so I will probably do it as a follow up PR

Reply:
implemented this